x/metadata/keeper/msg_server.go

```diff
@@ -65,17 +66,49 @@ func (k msgServer) AddScopeSpecification(
 ) (*types.MsgAddScopeSpecificationResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	// TODO (contract keeper class  methods to process request, keeper methods to record it)
+	existing, _ := k.GetScopeSpecification(ctx, msg.Specification.SpecificationId)
+	if err := k.ValidateScopeSpecUpdate(ctx, existing, *msg.Specification, msg.Signers); err != nil {
+		return nil, err
+	}
+
+	k.SetScopeSpecification(ctx, *msg.Specification)
 
 	ctx.EventManager().EmitEvent(
 		sdk.NewEvent(
 			sdk.EventTypeMessage,
 			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
-			sdk.NewAttribute(sdk.AttributeKeySender, ""),
+			sdk.NewAttribute(sdk.AttributeKeySender, strings.Join(msg.Signers, ",")),
```

Comment:
Not sure I have seen this done before... it makes sense.  I suspect this is a perfect example of why the upcoming typed event stuff in #88  will be so important though